plugins/simpleadminhacks-paper/src/main/java/com/programmerdan/minecraft/simpleadminhacks/hacks/BuildLimits.java

```diff
@@ -0,0 +1,65 @@
+package com.programmerdan.minecraft.simpleadminhacks.hacks;
+
+import com.programmerdan.minecraft.simpleadminhacks.SimpleAdminHacks;
+import com.programmerdan.minecraft.simpleadminhacks.configs.BuildLimitsConfig;
+import com.programmerdan.minecraft.simpleadminhacks.framework.SimpleHack;
+import com.programmerdan.minecraft.simpleadminhacks.framework.utilities.BuildLimit;
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.text.format.NamedTextColor;
+import org.bukkit.Location;
+import org.bukkit.configuration.ConfigurationSection;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.Listener;
+import org.bukkit.event.block.BlockPlaceEvent;
+import org.jetbrains.annotations.NotNull;
+
+public class BuildLimits extends SimpleHack<BuildLimitsConfig> implements Listener {
+	public BuildLimits(SimpleAdminHacks plugin, BuildLimitsConfig config) {
+		super(plugin, config);
+	}
+
+	public static BuildLimitsConfig generate (
+			final @NotNull SimpleAdminHacks plugin,
+			final @NotNull ConfigurationSection config){
+		return new BuildLimitsConfig(plugin, config);
+	}
+
+	@Override
+	public void onEnable(){
+		plugin.registerListener(this);
+	}
+
```

Comment:
Needs an `onDisable` override here to remove this listener, then the `if(!config.isEnabled()) return;` checks are not necessary since the event will only ever be called when the hack is enabled, because the listeners will only ever be registered upon hack enable.

Reply:
`Listener#unregisterListener` is depricated

Reply:
Use [`HandlerList.unregisterAll(this);`](https://github.com/CivMC/Civ/blob/c6278bd01319bbf55c415113efd89ba3cad61f2a/plugins/simpleadminhacks-paper/src/main/java/com/programmerdan/minecraft/simpleadminhacks/framework/BasicHack.java#L23), which is what BasicHack uses.